crates/types/src/blockchain/block.rs

```diff
@@ -80,6 +80,23 @@ impl Block<Transaction> {
         }
     }
 
+    /// Try creating a new full fuel block from a [`BlockHeader`] and
+    /// **previously executed** transactions.
+    /// This will fail if the transactions don't match the header.
+    pub fn try_from_executed(
+        header: BlockHeader,
+        mut transactions: Vec<Transaction>,
+    ) -> Option<Self> {
+        let transaction_ids: Vec<_> =
```

Comment:
Actually the `txRoot` is the merkle tree of the transaction bytes **not** the transaction ids. The reason is because the id is the same regardless of whether the transaction is executed or not but the merkle root needs to be based on executed transactions. This is a good reason why we should have a seperate type for executed vs zeroed transactions. @Voxelot the [spec](https://fuellabs.github.io/fuel-specs/master/protocol/block_header.html) doesn't make this clear.

Reply:
While we are awaiting a response from @Voxelot .

I made that conclusion based on the name of the variables=) If we really need to use `tx_bytes` instead of `tx_id`(transaction id is a `hash` of the `to_bytes`), then we need to update the naming. Because `transaction_ids` means `identifiers of transactions`.

Reply:
@Voxelot Can you comment on this?

Reply:
This code looks correct to me, we do need the tx bytes for the txroot. Although I agree that maybe a different name from transaction ids makes sense. Especially since this is just bytes that haven't had any additional hashing or anything else done to them, so maybe `transction_ids` -> `tx_bytes`?